pylint/checkers/unicode.py

```diff
@@ -0,0 +1,701 @@
+            f'use "{self.escaped}" instead.'
+        )
+
+    def human_code(self) -> str:
+        """Used to generate the human readable error message"""
+        return f"invalid-character-{self.name}"
+
+
+def line_length(line: _StrLike, codec: str) -> int:
+    """Get the length of a string like line as displayed in an editor"""
+    codec = _normalize_codec_name(codec)
+
+    if isinstance(line, bytes):
+        decoded = _remove_bom(line, codec).decode(codec, "replace")
+    else:
+        decoded = line
+
+    stripped = decoded.rstrip("\n")
+
+    if stripped != decoded:
+        stripped = stripped.rstrip("\r")
+
+    return len(stripped)
+
+
+def _map_positions_to_result(
+    line: _StrLike,
+    search_dict: Dict[_StrLike, _BadChar],
+    new_line: _StrLike,
+    byte_str_length: int = 1,
+) -> Dict[int, _BadChar]:
+    """Get all occurrences of search dict keys within line
+
+    Ignores Windows end of line and can handle bytes as well as string.
+    Also takes care of encodings for which the length of an encoded code point does not
+    default to 8 Bit.
+    """
+
+    result: Dict[int, _BadChar] = {}
+    ignore_pos: Optional[int]
+    for search_for, char in search_dict.items():
+        if search_for not in line:
+            continue
+
+        # Special Handling for Windows '\r\n'
+        if char.unescaped == "\r" and line.endswith(new_line):
+            ignore_pos = len(line) - 2 * byte_str_length
+        else:
+            ignore_pos = None
+
+        start: int = 0
+        pos: int = line.find(search_for, start)
+        while pos > 0:
+            if pos != ignore_pos:
+                # Calculate the
+                col = int(pos / byte_str_length)
+                result[col] = char
+            start = pos + 1
+            pos = line.find(search_for, start)
+
+    return result
+
+
+UNICODE_BOMS: Dict[str, bytes] = {
+    "utf-8": codecs.BOM_UTF8,
+    "utf-16": codecs.BOM_UTF16,
+    "utf-32": codecs.BOM_UTF32,
+    "utf-16le": codecs.BOM_UTF16_LE,
+    "utf-16be": codecs.BOM_UTF16_BE,
+    "utf-32le": codecs.BOM_UTF32_LE,
+    "utf-32be": codecs.BOM_UTF32_BE,
+}
+
+
+@lru_cache(maxsize=1)
+def get_bom_sorted() -> Dict[bytes, str]:
+    """Get all BOM sorted by size, biggest first"""
+    result = OrderedDict()
+    for codec, bom in sorted(
+        UNICODE_BOMS.items(), key=lambda x: len(x[1]), reverse=True
+    ):
+        if codec not in ("utf-16", "utf-32"):
+            # only use codecs with defined endian
+            result[bom] = codec
+    return result
+
+
+def _normalize_codec_name(codec: str) -> str:
+    """Make sure the codec name is always gives as defined in the BOM dict"""
+    codec = codec.lower()
+    return re.sub("utf[ -]?(8|16|32)[ -]?(le|be|)", r"utf-\1\2", codec)
+
+
+def _remove_bom(encoded: bytes, encoding: str) -> bytes:
+    """remove the bom if given from a line"""
+    encoding = _normalize_codec_name(encoding)
+    if not encoding.startswith("utf"):
+        return encoded
```

Comment:
This is uncovered. Are we able to write a test case for this or should we add `# pragma: no cover`?

Reply:
I can write a test case for this.
If I implement the check for UTF-8 ([see my note](https://github.com/PyCQA/pylint/pull/5311/files/835d1a9edab2aa9f37e238809a827f0bee4c8365#r777434368)), I would add a latin1 test case anyway.

This test case would cover this anyway. So first tell me if to continue with this check.